### PR TITLE
CGAZ bug squashin'

### PR DESCRIPTION
**Pull request by chapmanjacobd**
- fill gaps in countries that have no admin2 or admin1 with admin1 or admin0, respectively (I think this is what most people would expect for a data product like CGAZ. this is also the behavior of GADM)
- attempt to remove all empty columns (but it seems like mapshaper adds a few)
- rewrite code to be a little more concise

### CODE REVIEW

**Comment by DanRunfola** [comment]
Neat - never seen a WIP PR.  Let me know if you have any questions about my spaghetti code :)

**Comment by chapmanjacobd** [comment]
it's starting to look a bit better :) 

image
image

**Comment by chapmanjacobd** [comment]
I think those are all the changes I'd like to propose for now and when I have more time in the future I'll make a separate PR for trying to align the metadata a bit more

File: gbCGAZBuild.py [hunk 1]
```diff
@@ -1,366 +1,324 @@
...
+    g.to_file(DTA_A0Path, driver="GeoJSON")
+
+    # Get the ADM1 and ADM2 from geoBoundaries.
+    # There should be only RARE exceptions where ADM1 and ADM2 do not exist.
+    A0Path = gBPath + curISO + "/ADM0/geoBoundaries-" + curISO + "-ADM0.geojson"
+    A1Path = gBPath + curISO + "/ADM1/geoBoundaries-" + curISO + "-ADM1.geojson"
+    A2Path = gBPath + curISO + "/ADM2/geoBoundaries-" + curISO + "-ADM2.geojson"
+    # If they do not exist use the nearest layer to not have gaps in the world
+    if not os.path.isfile(A1Path):
+        A1Path = A0Path
```
Comment (chapmanjacobd) [inline gbCGAZBuild.py:184]: or should this be `DTA_A0Path` ?

**Comment by DanRunfola** [comment]
Got it - I'll try to review this ASAP.

File: gbCGAZBuild.py (hunk 1 above)
Comment (DanRunfola) [reply gbCGAZBuild.py:184]: A0Path is correct here.  Basically, if a ADM2 or ADM1 doesn't exist, we still gap-fill using ADM0 to give the contigious coverage.

**Review by DanRunfola — APPROVED** [review]
Just did a full implementation - much more elegant than what I had!

**Comment by DanRunfola** [comment]
This is a thing of beauty, many thanks.  Going to be pushing the new CGAZ in just a few minutes.

**Pull request merged**